tests/test_bundle.py

```diff
@@ -332,6 +338,58 @@ async def test_disable_ingress_auth(ops_test: OpsTest):
     )
 
 
+async def test_gateway_replicas_config(ops_test: OpsTest):
+    """Test changing the replicas config to 2, and then:
+    1. Assert the Deployment has the antiaffinity rule as expected.
+    2. Assert the new Pod was not scheduled due to only 1 Node being available.
+    """
+
+    replicas_value = "2"
+    await ops_test.model.applications[ISTIO_GATEWAY_APP_NAME].set_config(
+        {"replicas": replicas_value}
+    )
+    await ops_test.model.wait_for_idle(apps=[ISTIO_GATEWAY_APP_NAME], status="active", timeout=300)
+
+    client = lightkube.Client()
+
+    # Get the gateway Deployment
+    gateway_deployment = client.get(
+        Deployment, name="istio-ingressgateway-workload", namespace=ops_test.model_name
+    )
+
+    # Assert the Deployment has the correct antiaffinity rule
+    assert (
+        str(
+            gateway_deployment.spec.template.spec.affinity.podAntiAffinity.requiredDuringSchedulingIgnoredDuringExecution[  # noqa E501
+                0
+            ].labelSelector.matchExpressions[
+                0
+            ]
+        )
+        == EXPECTED_LABEL_SELECTOR
+    )
```

Comment:
Hey @NohaIhab I would've expected this to be tested in the unit test as well. Is there any reason to keep it here?